airflow/www_rbac/views.py

```diff
@@ -2567,6 +2600,32 @@ def action_set_success(self, drs, session=None):
             flash('Failed to set state', 'error')
         return redirect(self.get_default_url())
 
+    @action('set_canceled', "Set state to 'canceled'",
+            "All task instances for the selected DAG Runs will be canceled. Are you sure?",
+            single=False)
+    @provide_session
+    def action_set_canceled(self, drs, session=None):
+        try:
+            DR = models.DagRun
+            dagrun_count = 0
+            task_instance_count = 0
+            dirty_ids = []
+            for dr in session.query(DR).filter(
+                DR.id.in_([dagrun.id for dagrun in drs])).all():
+                dirty_ids.append(dr.dag_id)
+                dag = dagbag.get_dag(dr.dag_id)
+                dagrun_count += 1
+                task_instance_count += dag.cancel(
+                    start_date=dr.execution_date,
+                    end_date=dr.execution_date,
+                    include_subdags=True)
+            flash(
+                "{count} dag runs and {altered_ti_count} task instances "
+                "were set to success".format(count=dagrun_count, altered_ti_count=task_instance_count))
```

Comment:
maybe put this in a `finally` block in case there is an exception before the loop finishes

Reply:
Done!